Src/AutoFakeItEasy/PropertyRule.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using AutoFixture.Kernel;
+using FakeItEasy.Core;
+
+namespace AutoFixture.AutoFakeItEasy
+{
+    /// <summary>
+    /// A rule that intercepts property calls. Set values will be saved and
+    /// provided as the return value from the matching get methods when the latter are called.
+    /// If a get method is called before the corresponding set, the return value will
+    /// be resolved from the fixture and similarly saved for later.
+    /// </summary>
+    internal class PropertyRule : IFakeObjectCallRule
+    {
+        private readonly ISpecimenContext context;
+
+        private readonly ConcurrentDictionary<PropertyCall, PropertyReturnValue> behaviors =
+            new ConcurrentDictionary<PropertyCall, PropertyReturnValue>();
+
+        public PropertyRule(ISpecimenContext context)
+        {
+            this.context = context;
+        }
+
+        public int? NumberOfTimesToCall { get; } = null;
+
+        public bool IsApplicableTo(IFakeObjectCall fakeObjectCall)
+        {
+            return fakeObjectCall != null && IsProperty(fakeObjectCall.Method);
+        }
+
+        public void Apply(IInterceptedFakeObjectCall interceptedfakeObjectCall)
+        {
+            if (interceptedfakeObjectCall == null) throw new ArgumentNullException(nameof(interceptedfakeObjectCall));
+
+            var fakeObjectCall = new FakeObjectCall(interceptedfakeObjectCall);
+            var propertyCall = new PropertyCall(fakeObjectCall);
+            var methodReturnType = fakeObjectCall.Method.ReturnType;
+            if (IsSetter(fakeObjectCall))
+            {
+                this.behaviors[propertyCall] = new PropertyReturnValue(fakeObjectCall.Arguments.Last());
+            }
+            else
+            {
+                var returnValueAction = this.behaviors.GetOrAdd(
+                    propertyCall,
+                    _ => new PropertyReturnValue(this.context.Resolve(methodReturnType)));
+                returnValueAction.ApplyToCall(fakeObjectCall);
+            }
+        }
+
+        private static bool IsProperty(MethodInfo method) =>
+            method.IsSpecialName && (method.Name.StartsWith("get_", StringComparison.Ordinal) ||
+                                     method.Name.StartsWith("set_", StringComparison.Ordinal));
+
+        private static bool IsSetter(FakeObjectCall fakeObjectCall) =>
+            fakeObjectCall.Method.ReturnType == typeof(void);
```

Comment:
Given that this method is very tricky, please duplicate the `method.IsSpecialName && method.Name.StartsWith("set_", StringComparison.Ordinal)` check here to make it more robust. Otherwise, later somebody might miss to verify for `IsProperty` first and we'll get very weird bugs 😅